### DISCUSSION TITLE
When to remove old data passed parameter keep setting?

### DISCUSSION THREAD

**Discussion by caojianwudong** [taosdata/TDengine#4589] [General]
我尝试设置了keep和days都为1，目的是实现数据库每过一天数据库里的内容删除一次，但是我测试后发现，两天后数据库中还存储着之前的数据，只有在关闭，再重启了taosd的服务后才删除了数据，请问这个是什么原因导致的

**Comment by stephenkgu** [comment]
1, As you tried, restart taosd;
2, Insert more data to trigger data file saving to the disk, then old data passed keep setting would be erased.

**Comment by caojianwudong** [comment]
了解了，重启确实可以触发过期的数据被删除，插入数据触发删除，这个需要插入多少数据才可以触发呢

**Reply by yu285** [reply]
当每个vnode写入了 三分之1的 cache*blocks 大小，就会触发。
